modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,148 +558,187 @@ sub print_diagram {
                     $need_write = 1;
                 }
+                push @{$responseStructure->{Tweaks}}, $tweakStructure;
             }
 
+
         } elsif($tweak=~/^analysis\[([^\]]+)\]\.(wait_for|flow_into)(\?|#|\+?=(.+))$/) {
+
             my ($analyses_pattern, $attrib_name, $operation, $new_value_str) = ($1, $2, $3, $4);
             $operation=~/^(\?|#|\+?=)/;
             my $operator = $1;
 
             my $analyses = $self->collection_of( 'Analysis' )->find_all_by_pattern( $analyses_pattern );
-            print "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
+            push @response, "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
 
             my $new_value = destringify( $new_value_str );
 
             foreach my $analysis (@$analyses) {
-
+                my $tweakStructure;
+                $tweakStructure->{Object}->{Type} = "Analysis";
+                $tweakStructure->{Action} = $self->{ACTION}->{substr($operator, 0, 1)};
                 my $analysis_name = $analysis->logic_name;
-
+                $tweakStructure->{Object}->{Id} = $analysis->dbID + 0;
+                $tweakStructure->{Object}->{Name} = $analysis->logic_name;
+                $tweakStructure->{Return}->{Field} = $attrib_name;
                 if( $attrib_name eq 'wait_for' ) {
-
                     my $cr_collection   = $self->collection_of( 'AnalysisCtrlRule' );
                     my $acr_collection  = $analysis->control_rules_collection;
-
+                    $tweakStructure->{Return}->{OldValue} = [map { $_->condition_analysis_url } @$acr_collection];
                     if($operator eq '?') {
-                        print "Tweak.Show    \tanalysis[$analysis_name].wait_for ::\t[".join(', ', map { $_->condition_analysis_url } @$acr_collection )."]\n";
+                        $tweakStructure->{Return}->{NewValue} = $tweakStructure->{Return}->{OldValue};
+                        push @response, "Tweak.Show  \tanalysis[$analysis_name].wait_for ::\t[".join(', ', map { $_->condition_analysis_url } @$acr_collection )."]\n";
                     }
 
                     if($operator eq '#' or $operator eq '=') {     # delete the existing rules
+                        $tweakStructure->{Return}->{NewValue} = undef;
                         foreach my $c_rule ( @$acr_collection ) {
                             $cr_collection->forget_and_mark_for_deletion( $c_rule );
                             $need_write = 1;
 
-                            print "Tweak.Deleting\t".$c_rule->toString." --> (missing value)\n";
+                            push @response, "Tweak.Deleting\t".$c_rule->toString." --> (missing value)\n";
                         }
                     }
 
                     if($operator eq '=' or $operator eq '+=') {     # create new rules
+                        $tweakStructure->{Return}->{NewValue} = $new_value;
```

Comment:
I think I've already reported this. When the action is `+=`, `$new_value` is the stuff that is _added_. I would expect `NewValue` to contains the complete final value, i.e. what was already there + what's been added

This is what we currently get:
```
$ tweak_pipeline.pl -url $(mysql-ens-compara-prod-6-ensadmin details url muffato_test_plants_goc_96) -tweak 'analysis[2].wait_for+=aaa' -json | json_pp
```
```json
{
   "URL" : "'mysql://ensadmin:${_EHIVE_HIDDEN_PASS}@mysql-ens-compara-prod-6.ebi.ac.uk:4616/muffato_test_plants_goc_96'",
   "Tweaks" : [
      {
         "Object" : {
            "Name" : "backbone_fire_clustering",
            "Id" : 2,
            "Type" : "Analysis"
         },
         "Action" : "SET",
         "Return" : {
            "Field" : "wait_for",
            "NewValue" : "aaa",
            "OldValue" : [
               "backbone_fire_clustering",
               "backbone_fire_db_prepare"
            ]
         }
      }
   ]
}
```